### DISCUSSION TITLE
OpenID Connect login fails: user authentication ... already exists for ..., tried to assign to ...

### DISCUSSION THREAD

**Discussion by decentral1se** [outline/outline#3213] [Self Hosting] [answered]
When trying to log in with OpenID Connect setup:

> {"error":"User authentication 76e22b20-a97a-4a26-a760-5e9efb9bbaa4 already exists for a3774944-3d45-4dd7-9c8a-4f5278cc9653, tried to assign to 032aa1e5-0cf0-4cdc-8386-17aedd8b62fa","stack":"Error: User authentication 76e22b20-a97a-4a26-a760-5e9efb9bbaa4 already exists for a3774944-3d45-4dd7-9c8a-4f5278cc9653, tried to assign to 032aa1e5-0cf0-4cdc-8386-17aedd8b62fa\n    at userCreator (/opt/outline/build/server/commands/userCreator.js:47:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async accountProvisioner (/opt/outline/build/server/commands/accountProvisioner.js:57:20)\n    at async OAuth2Strategy._verify (/opt/outline/build/server/routes/auth/providers/oidc.js:93:22)","level":"error","message":"Error during authentication"}

We migrated from v0.60.x to v0.62.x. We had previously run into https://github.com/outline/outline/issues/2678 and had been testing with https://github.com/outline/outline/pull/2659 until it was closed. We then upgraded to latest and it does work for most users, they can log in but for 2 of our users, they cannot log in.

**To Reproduce**
1. Setup OpenID Connect provider
2. Log in

**Expected behavior**
You can log in.

**Screenshots**
N/A

**Outline (please complete the following information):**
- Install: self-hosted
- Version: v0.62

**Desktop (please complete the following information):**
 - OS: Debian
 - Browser Chrome
 - Version 98.0.4758.102

**Mobile (please complete the following information):**
 - Device: N/A
 - OS: N/A
 - Browser N/A

**Comment by decentral1se** [comment, marked as answer]
Omg everything i've posted as an issue has been moved to discussions across several projects in the last month :roll_eyes: 

The new wont-fix.

```
outline=# DELETE FROM users WHERE  username = 'foo';
ERROR:  update or delete on table "users" violates foreign key constraint "revisions_userId_fkey" on table "revisions"
DETAIL:  Key (id)=(d0446a00-7879-4d3b-add5-f93ed4f78da1) is still referenced from table "revisions".
DELETE FROM revisions WHERE userId = "d0446a00-7879-4d3b-add5-f93ed4f78da1";
```

"Works"

**Comment by tommoor** [comment]
Hi @decentral1se – moved to discussion because the problem is with your database data in a self-hosted instance caused by running a fork of the codebase. I'm glad you were able to workaround it and hopefully your answer here will help future folks :)

**Comment by decentral1se** [comment]
Still running into variations of this, a more robust fix seems to be abusing the invite mechanism in https://github.com/outline/outline/blob/main/server/commands/userCreator.ts#L87-L143 and setting the "last active at" to null for all users who are facing problems logging in, this will then re-map the authentication providers correctly on re-log-in resulting in a working entry in the user_authentications table.
